net/wgpeerselector/files/usr/bin/wgpeerselector

```diff
@@ -0,0 +1,501 @@
+#!/usr/bin/lua
+local unistd = require 'posix.unistd'
+local syslog = require 'posix.syslog'
+local grp = require 'posix.grp'
+local sys_sock = require 'posix.sys.socket'
+local sys_stat = require 'posix.sys.stat'
+local signal = require 'posix.signal'
+local time = require 'posix.time'
+local uloop = require 'uloop'
+local ubus = require 'ubus'
+local uci = require('simple-uci').cursor()
+
+local function log(level, message)
+	if _G.is_verbose then
+		local prefix = {
+			[syslog.LOG_ERR]  = "Error: ",
+			[syslog.LOG_CRIT] = "Critical: ",
+			[syslog.LOG_INFO] = "Info: ",
+			[syslog.LOG_WARNING] = "Warning: ",
+		}
+		io.stderr:write(prefix[level]..message .. '\n')
+	end
+
+	syslog.syslog(level, message)
+end
+
+local function execute(command)
+	local f = io.popen(command)
+	if not f then
+		return nil
+	end
+
+	local data = f:read('*a')
+	f:close()
+	return data
+end
+
+local function sleep(secs)
+	-- This function calls uloop.run() for 'secs' seconds, so we can handle i/o
+	-- during that time. But be aware, that an fd should be registered to uloop,
+	-- because otherwise this function uses 100% cpu. This might be a bug in
+	-- uloop.
+	local timer = uloop.timer(function() uloop.cancel() end)
+	timer:set(secs*1000)
+	uloop.run()
```

Comment:
Why not just use [`unistd.sleep`](https://luaposix.github.io/luaposix/modules/posix.unistd.html#sleep)? Anything special about uloop that makes it a better candidate?

Reply:
Yes. That is exactly why we use it here. It allows us to serve the ubus api even when we are "asleep".